app/code/Magento/Store/App/Request/PathInfoProcessor.php

```diff
@@ -44,7 +44,9 @@ public function process(\Magento\Framework\App\RequestInterface $request, $pathI
 
         if ($store->isUseStoreInUrl()) {
             if (!$request->isDirectAccessFrontendName($storeCode) && $storeCode != Store::ADMIN_CODE) {
-                $this->storeManager->setCurrentStore($storeCode);
+                if ($storeCode !== '') {
```

Comment:
There is ` } elseif (!empty($storeCode)) {` condition in the bottom, can the whole logic be simplified maybe?

Reply:
@orlangur finally I added `setCurrentStore` from `$store->getCode()` and I adjust tests for this logic